auction-server/src/auction/service/verification.rs

```diff
@@ -498,6 +499,64 @@ impl Service<Svm> {
         }
     }
 
+    fn validate_swap_transaction_instructions(
+        &self,
+        tx: &VersionedTransaction,
+    ) -> Result<(), RestError> {
+        tx.message
+            .instructions()
+            .iter()
+            .enumerate()
+            .try_for_each(|(index, ix)| {
+                self.validate_swap_transaction_instruction(tx.message.static_account_keys(), ix)
+                    .map_err(|e| {
+                        RestError::BadParameters(format!(
+                            "Invalid instruction at index {}: {:?}",
+                            index, e
+                        ))
+                    })
+            })?;
+
+        Ok(())
+    }
+
+    fn validate_swap_transaction_instruction(
+        &self,
+        accounts: &[Pubkey],
+        ix: &CompiledInstruction,
+    ) -> Result<(), RestError> {
+        let program_id = accounts
+            .get(ix.program_id_index as usize)
+            .ok_or_else(|| RestError::BadParameters("Invalid program id index".to_string()))?;
+
+        if *program_id == compute_budget::id() {
+            Ok(())
+        } else if *program_id == self.config.chain_config.associated_token_program_id {
+            let ix_parsed =
+                AssociatedTokenAccountInstruction::try_from_slice(&ix.data).map_err(|e| {
+                    RestError::BadParameters(format!(
+                        "Invalid associated token account instruction: {}",
+                        e
+                    ))
+                })?;
+            match ix_parsed {
+                AssociatedTokenAccountInstruction::Create => Ok(()),
+                AssociatedTokenAccountInstruction::CreateIdempotent => Ok(()),
+                _ => Err(RestError::BadParameters(format!(
+                    "Invalid associated token account instruction: {:?}",
+                    ix_parsed
+                ))),
+            }
+        } else if *program_id == self.config.chain_config.express_relay.program_id {
```

Comment:
would this work as a match statement?

Reply:
originally had it as that but it was a bit weird. `match` doesn't work with non-constant expressions, so you have to define the expressions outside of the match statement. i can make that change and we can choose whichever way you prefer